### PR TITLE
feat: WIP-create-password-aftre member-invite

### PR DESCRIPTION
**Pull request by arpita-20**
- before, creating a new member, password auto-generated password will be sent to the invited member's mail
- update: now, instead of auto-generation of password , user'll get the link on his mail, from where user can create a password

### CODE REVIEW

File: app/components/login/CreatePassword.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  checkErrorStatus,
+}: {
+  checkErrorStatus: boolean
+}) => {
+  const [enterPassword, setEnterPassword] = useState('')
+  const [reEnterPassword, setReEnterPassword] = useState('')
+  const { t } = useTranslation()
+  const inputFieldsPropsforEnterPassword = [
+    {
+      label: '',
```
Comment (yash6102000) [inline app/components/login/CreatePassword.tsx:17]: why label is not given

File: app/components/login/CreatePassword.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react'
```
Comment (yash6102000) [inline app/components/login/CreatePassword.tsx:1]: just like other pages header should be sticky

File: app/components/login/CreatePassword.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      errorId: 'email-error',
+      onChange: function (event: any) {
+        setReEnterPassword(event?.target.value)
+      },
+    },
+  ]
+
+  return (
+    <div className="flex flex-1 items-center justify-center bg-gray-50">
+      <div className="flex w-full max-w-454 flex-col gap-12 rounded-md border border-gray-50 bg-white px-12 pb-12  text-center drop-shadow-xl">
```
Comment (yash6102000) [inline app/components/login/CreatePassword.tsx:47]: no need to use padding bottom and padding top ..use padding-12 instead

**Comment by richa-copods** [comment]
Please fix cypress test cases

File: app/models/sendgrid.servers.ts [hunk 4]
```diff
@@ -49,14 +49,15 @@ export async function sendMail(
...
-          <p style="text-align:left;color:#4B5563;margin:0;font-family:'Poppins', sans-serif;">
-            Password: <span style="font-weight:600">${password}</span>
+
+        </div>        
+        <p style="text-align:center;color:#4B5563;margin:0;font-family:'Poppins', sans-serif;margin-top:10px">     
+          <a href="${passwordGenerationLink}" style="color: #353988;">"${passwordGenerationLink}"</a>
+        </p>
+        <br>
+         <p style="text-align:center;color:#4B5563;margin:0;font-family:'Poppins', sans-serif;">
+             Use above link to generate your password
```
[the remark this answers is not in the document]
Comment (arpita-20) [reply app/models/sendgrid.servers.ts:59]: @yash6102000 this is  the sendgrid file. please check before providing comment

File: app/routes/members.$userId.create-password.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export const loader: LoaderFunction = async ({ request, params }) => {
+  invariant(params.userId, 'testId not found')
+  return null
+}
+export const action: ActionFunction = async ({ request, params }) => {
+  const formData = await request.formData()
+  const userId = params.userId
+  const enteredPassword = formData.get('enterPassword')
+  const reEnteredPassword = formData.get('reEnterPassword')
+  let res = null
```
[the remark this answers is not in the document]
Comment (arpita-20) [reply app/routes/members.$userId.create-password.tsx:20]: it ihas been used later. Please check @yash6102000

File: app/components/login/CreatePassword.tsx (hunk 2 above)
Comment (arpita-20) [reply app/components/login/CreatePassword.tsx:1]: it is correct @yash6102000

File: app/routes/members.$userId.create-password.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  const formData = await request.formData()
+  const userId = params.userId
+  const enteredPassword = formData.get('enterPassword')
+  const reEnteredPassword = formData.get('reEnterPassword')
+  if (enteredPassword === reEnteredPassword) {
+    await createPasswordOfUser(userId as string, enteredPassword as string)
+    return redirect(routes.signIn)
+  }
+  if (enteredPassword !== reEnteredPassword) {
+    const res = 'not equal'
```
Comment (richa-copods) [inline app/routes/members.$userId.create-password.tsx:25]: No need of creating this extra variable

**Review by richa-copods — DISMISSED** [review]

File: public/locales/en/index.json [hunk 7]
```diff
@@ -51,7 +51,10 @@
     "totalQuestion": "Total Questions",
     "srNo": "Sr.No",
     "verify": "Verify",
-    "proceed": "Proceed"
+    "proceed": "Proceed",
```
Comment (anuragP04) [inline public/locales/en/index.json:54]: instead of creating proceed, we can use submit

File: public/locales/en/index.json [hunk 8]
```diff
@@ -51,7 +51,10 @@
...
+    "enterPassword": "Enter Password",
```
Comment (anuragP04) [inline public/locales/en/index.json:55]: make it password, which is already present, not need to create again

File: public/locales/en/index.json [hunk 9]
```diff
@@ -51,7 +51,10 @@
...
+    "reEnterPassword": "Re-Enter Password",
```
Comment (anuragP04) [inline public/locales/en/index.json:56]: make it confirm password

File: public/locales/en/index.json [hunk 10]
```diff
@@ -156,7 +159,8 @@
     "otpSent": "Otp sent. Please check your email",
     "correctOtp": "Please enter correct OTP",
     "resendPasswordSuccess": "New password has been sent to email successfully",
-    "resendPasswordError": "Account not found. Please enter valid email"
+    "resendPasswordError": "Account not found. Please enter valid email",
+    "enteredReenteredPassword": "Entered and Re-Entered Password should be equal."
```
Comment (anuragP04) [inline public/locales/en/index.json:163]: Password and Confirm password should be same.

File: cypress/e2e/members-test/members.cy.ts [hunk 11]
```diff
@@ -155,4 +155,105 @@ describe('Test for members', () => {
     })
     return false
   })
+  // it('create-password page contain correct header', () => {
```
Comment (anuragP04) [inline cypress/e2e/members-test/members.cy.ts:158]: why these tests are commented

File: cypress/e2e/members-test/members.cy.ts [hunk 12]
```diff
@@ -155,4 +155,105 @@ describe('Test for members', () => {
...
+  //   cy.visit(
```
Comment (anuragP04) [inline cypress/e2e/members-test/members.cy.ts:159]: create a test for invalid id password page

File: cypress/e2e/forgot-password/ForgotPassword.cy.ts [hunk 13]
```diff
@@ -48,10 +48,5 @@ describe('Forgot password', () => {
       .clear()
       .type(memberEmail)
     cy.get('#reset-password').click()
-    cy.get('.Toastify__toast').should(
```
Comment (anuragP04) [inline cypress/e2e/forgot-password/ForgotPassword.cy.ts:51]: why this assertion is removed?

File: app/models/sendgrid.servers.ts [hunk 14]
```diff
@@ -49,14 +49,15 @@ export async function sendMail(
           <p style="text-align:left;color:#4B5563;margin:0;font-family:'Poppins', sans-serif;">
             Email: <span style="font-weight:600">${email}</span>
           </p>
-          <p style="text-align:left;color:#4B5563;margin:0;font-family:'Poppins', sans-serif;">
-            Password: <span style="font-weight:600">${password}</span>
+
```
Comment (anuragP04) [inline app/models/sendgrid.servers.ts:52]: `
You have been added as Admin on K - Quiz.
Please use below link to create your first password.

${link}
`
Screenshot 2022-10-31 at 11 44 42 AM

File: app/interface/Interface.tsx [hunk 15]
```diff
@@ -2,7 +2,7 @@ import type React from 'react'
 
 export interface InputFieldProps {
   name: string
-  label: string
+  label?: string
```
Comment (anuragP04) [inline app/interface/Interface.tsx:5]: why we have made label as optional?

File: app/routes/members.$userId.create-password.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+import Header from '~/components/assessment/Header'
+import CreatePassword from '~/components/login/CreatePassword'
+import { redirect } from '@remix-run/node'
+import { useActionData } from '@remix-run/react'
+import type { LoaderFunction, ActionFunction } from '@remix-run/server-runtime'
+import invariant from 'tiny-invariant'
+import { createPasswordOfUser, getAllUsers } from '~/models/user.server'
+import { routes } from '~/constants/route.constants'
+
+export const loader: LoaderFunction = async ({ request, params }) => {
```
Comment (anuragP04) [inline app/routes/members.$userId.create-password.tsx:11]: add a condition if user have already created the password, show him, you have already created password.

File: app/routes/members.$userId.create-password.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+  const getUsers = await getAllUsers()
```
Comment (anuragP04) [inline app/routes/members.$userId.create-password.tsx:12]: add condition/check if userid in link is not present in db, : invalid/broken link

File: cypress/e2e/forgot-password/ForgotPassword.cy.ts (hunk 13 above)
Comment (anuragP04) [reply cypress/e2e/forgot-password/ForgotPassword.cy.ts:51]: havn't resolved this

File: app/components/login/CreatePassword.tsx [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+              })}
+            </div>
+            <Button
+              tabIndex={0}
+              id="reset-password"
+              varient="primary-solid"
+              type="submit"
+              className="h-11 "
+              title="proceed"
+              buttonText="Proceed"
```
Comment (anuragP04) [inline app/components/login/CreatePassword.tsx:75]: why this is hardcoded, use i18next variable submit here

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**